platform/macos/detect.py

```diff
@@ -239,8 +239,17 @@ def configure(env: "SConsEnvironment"):
 
     env.Append(LINKFLAGS=["-rpath", "@executable_path/../Frameworks", "-rpath", "@executable_path"])
 
+    if env["metal"] and env["arch"] != "arm64":
+        print("Metal is only supported on arm64.")
+        sys.exit(255)
+
+    if env["metal"]:
```

Comment:
Might be better to use arch check here, and remove `sys.exit()` if before, this way it can be built for both x86 and arm with the same arguments. And can be enabled by default.

```suggestion
    if env["metal"] and env["arch"] == "arm64":
```

Reply:
Sounds like a plan.

Reply:
Done – now builds `metal` by default, and does not exit for non-arm64 builds